http_client.go

```diff
@@ -318,6 +319,14 @@ func (c *HTTPClient) Send(data []byte) (response []byte, err error) {
 		}
 	}
 
+	if (c.config.CloseConnectionOnResponseStatus > 0) {
```

Comment:
I thought this'd be better as it doesn't change current behavior and others might want this for other codes like 503 or whatever...But it's your call.